Source/WebCore/Modules/webaudio/AudioBuffer.cpp

```diff
@@ -117,13 +119,14 @@ AudioBuffer::AudioBuffer(AudioBus& bus)
     : m_sampleRate(bus.sampleRate())
     , m_originalLength(bus.length())
 {
-    if (static_cast<uint64_t>(m_originalLength) > s_maxLength) {
+    unsigned numberOfChannels = bus.numberOfChannels();
+    auto totalLength = CheckedUint64(m_originalLength) * numberOfChannels;
+    if (totalLength.hasOverflowed() || totalLength.value() > s_maxLength || static_cast<uint64_t>(m_originalLength) > s_maxChannelLength) {
```

Comment:
It would be good to put this in a shared function rather than duplication this logic.